app/assets/javascripts/editor.ts

```diff
@@ -0,0 +1,221 @@
+import { closeBrackets, closeBracketsKeymap } from "@codemirror/autocomplete";
+import { defaultKeymap, history, historyKeymap } from "@codemirror/commands";
+import {
+    bracketMatching,
+    foldGutter,
+    foldKeymap,
+    HighlightStyle,
+    indentOnInput, LanguageDescription,
+    syntaxHighlighting
+} from "@codemirror/language";
+import { languages } from "@codemirror/language-data";
+import {
+    drawSelection,
+    dropCursor,
+    EditorView,
+    highlightActiveLine,
+    highlightActiveLineGutter,
+    highlightSpecialChars,
+    keymap,
+    lineNumbers
+} from "@codemirror/view";
+import { tags } from "@lezer/highlight";
+import { Extension } from "@codemirror/state";
+
+declare type EditorEventHandler = (event: FocusEvent, view: EditorView) => boolean | void;
+
+
+// A custom theme for CodeMirror that applies the same CSS as Rouge does,
+// meaning we can use our existing themes.
+const rougeStyle = HighlightStyle.define([
+    { tag: tags.comment, class: "c" },
+    { tag: tags.lineComment, class: "c" },
+    { tag: tags.blockComment, class: "cm" },
+    { tag: tags.docComment, class: "cs" },
+    { tag: tags.name, class: "n" },
+    { tag: tags.variableName, class: "nv" },
```

Comment:
```suggestion
    { tag: tags.variableName, class: "n" },
```

It bothered me that a lot of code was italic, which was not the case in the rouge code rendering. I found that rouge uses `n` or `nf` instead of `nv` as class for the same variables.

Another solution for this might be to remove the italic styling from `nv`

eg:
![image](https://github.com/dodona-edu/dodona/assets/21177904/59ed6598-93b0-4efd-ad27-e7b2415a939d)
![image](https://github.com/dodona-edu/dodona/assets/21177904/1d237ff9-d1ac-45e9-91a7-c1a82e9f4d0b)
vs:
![image](https://github.com/dodona-edu/dodona/assets/21177904/f5f68c74-230e-44d4-ac85-c4b6eafbcfce)
![image](https://github.com/dodona-edu/dodona/assets/21177904/12755f13-5f7d-49e5-a6a1-f07d49614521)

Reply:
I removed the italic from `nv`, as it felt like a better solution than to make the css less specific.